components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -572,22 +585,21 @@ class CLPConfig(BaseModel):
     _os_release_file_path: pathlib.Path = PrivateAttr(default=OS_RELEASE_FILE_PATH)
 
     def make_config_paths_absolute(self, clp_home: pathlib.Path):
-        self.input_logs_directory = make_config_path_absolute(clp_home, self.input_logs_directory)
+        if StorageType.FS == self.logs_input.type:
+            self.logs_input.make_config_paths_absolute(clp_home)
         self.credentials_file_path = make_config_path_absolute(clp_home, self.credentials_file_path)
         self.archive_output.storage.make_config_paths_absolute(clp_home)
         self.stream_output.storage.make_config_paths_absolute(clp_home)
         self.data_directory = make_config_path_absolute(clp_home, self.data_directory)
         self.logs_directory = make_config_path_absolute(clp_home, self.logs_directory)
         self._os_release_file_path = make_config_path_absolute(clp_home, self._os_release_file_path)
 
-    def validate_input_logs_dir(self):
-        # NOTE: This can't be a pydantic validator since input_logs_dir might be a package-relative
-        # path that will only be resolved after pydantic validation
-        input_logs_dir = self.input_logs_directory
-        if not input_logs_dir.exists():
-            raise ValueError(f"input_logs_directory '{input_logs_dir}' doesn't exist.")
-        if not input_logs_dir.is_dir():
-            raise ValueError(f"input_logs_directory '{input_logs_dir}' is not a directory.")
+    def validate_logs_input_config(self):
+        if StorageType.FS == self.logs_input.type:
+            try:
+                validate_path_could_be_dir(self.logs_input.directory)
```

Comment:
Note to myself and Nit: The difference here is that we don't enforce the directory to exist. Need to think if there's way to deal with the inconsistent behavior

Reply:
Thinking about this again, let's add an extra boolean argument, such as "require_exists" to indicate if the directory must exist.

This will ensure the check behavior for worker's input log directory stays unchanged after this PR.